### ISSUE TITLE
Support for reversing animation at any point in time

### ISSUE THREAD

**Issue by mic-code** [AnnulusGames/LitMotion#187]
Related to https://github.com/AnnulusGames/LitMotion/pull/97

This is the simplest code I come up with for doing animation interrupt with the current API
Which is too clunky to use, it require two opposite tween definition.
Is there plan for supporting this use case better?

Basically a "reverse" function in additional to the current "play" "stop" "pause" "cancel"

```C#
using LitMotion;
using LitMotion.Extensions;
using NaughtyAttributes;
using UnityEngine;

public class Test : MonoBehaviour
{
    public Transform target;

    bool isOn;
    MotionHandle handle;

    [Button]
    public void Toggle()
    {
        isOn = !isOn;

        const float duration = 2;
        var start = Vector3.zero;
        var end = Vector3.up;

        double time = 0;
        if (handle != MotionHandle.None && handle.IsActive() && handle.IsPlaying())
        {
            handle.Cancel();
            time = duration - handle.Time;
        }

        if (isOn)
        {
            handle = LMotion.Create(start, end, duration)
                .BindToPosition(target);
        }
        else
        {
            handle = LMotion.Create(end, start, duration)
                .BindToPosition(target);
        }

        handle.Time = time;
    }
}
```

**Comment by eefan000** [comment]
I think we need methods similar to PlayBackwards and PlayForward in DoTween?
The MotionHandle in DoTween can remain valid even after a single playback, allowing for easy access to PlayBackwards and PlayForward methods as needed, making it convenient for the aforementioned use cases

**Comment by AnnulusGames** [comment]
This can be achieved by simply setting `handle.PlaybackSpeed` to a negative value (requires v2.0.0 or later).
No additional methods are required.

**Comment by eefan000** [comment]
> This can be achieved by simply setting `handle.PlaybackSpeed` to a negative value (requires v2.0.0 or later). No additional methods are required.

Can it still be used after the playback ends? Will the handle automatically recycle after playback is complete?
For example, play forward first, and when the playback progress reaches 1, use handle PlaybackSpeed Reverse Play

**Comment by AnnulusGames** [comment]
By using `handle.Preserve()`, which was added in v2.0.0, you can hold the `MotionHandle` even after playback has ended. (This is equivalent to DOTween's `SetAutoKill(false)`)

**Comment by eefan000** [comment]
I used version 1.0 before, but due to the lack of many APIs, I switched to primeweek. Now I plan to switch back! Especially LitMotion Animation， I originally planned to make one for Primetweek myself.
Can the AnimationMode be changed to a non global setting, similar to FlowType in [Animation Sequencer](https://github.com/brunomikoski/Animation-Sequencer)?

**Comment by AnnulusGames** [comment]
Currently, it is not possible to set the AnimationMode individually. I am considering adding this feature in the next version.

However, when creating complex animations with LitMotion Animation, I recommend splitting the animation into several LitMotion Animations and combining them with Play LitMotion Animation.

**Comment by eefan000** [comment]
Just some slightly complex small animations, which are more common when controlling multiple objects.
For example, object A rotates and scales, and object B also rotates and scales, but the animation order of objects A and B is different.
I would use UnityAnimation or Timeline+TweenPlayables for more complex animations

**Comment by AnnulusGames** [comment]
Yes, I agree. LitMotion Animation is not suitable for complex animations.

This is not related to the topic of this issue, so I will close this for now.

**Comment by micology-code** [comment]
Thanks, that's good to know there's a way to do this, I'll try it
Does this work for the sequence api as well?

**Comment by AnnulusGames** [comment]
Yes, it should work fine.
